test/e2etest/kubetest/results_evaluator.go

```diff
@@ -0,0 +1,177 @@
+func Analyze(kubetestResultsPath string) Summary {
+	log.Info("Analyze e2e.log and junit.xml files")
+	e2eLogFilePaths := util.GetFilesByPattern(kubetestResultsPath, E2eLogFileNamePattern)
+	summary := analyzeE2eLogs(e2eLogFilePaths)
+	junitXMLFilePaths := util.GetFilesByPattern(kubetestResultsPath, JunitXmlFileNamePattern)
+	analyzeJunitXMLs(junitXMLFilePaths, summary.TestsuiteDuration)
+	log.Infof("Check out result files in %s", kubetestResultsPath)
+	return summary
+}
+
+func analyzeJunitXMLs(junitXMLFilePaths []string, durationSec int) {
+	var mergedJunitXmlResult = &JunitXMLResult{FailedTests: 0, ExecutedTests: 0, DurationFloat: 0, SuccessfulTests: 0, DurationInt: durationSec}
+	testcaseNameToTestcase := make(map[string]TestcaseResult)
+	for _, junitXMLPath := range junitXMLFilePaths {
+		file, err := os.Open(junitXMLPath)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer file.Close()
+		junitXml, err := UnmarshalJunitXMLResult(file.Name())
+		if err != nil {
+			log.Fatal(errors.Wrapf(err, "Couldn't unmarshal %s", file.Name()))
+		}
+		mergedJunitXmlResult.FailedTests += junitXml.FailedTests
+		mergedJunitXmlResult.ExecutedTests += junitXml.ExecutedTests
+		mergedJunitXmlResult.SuccessfulTests += junitXml.SuccessfulTests
+		for _, testcase := range junitXml.Testcases {
+			if testcase.Skipped {
+				if _, ok := testcaseNameToTestcase[testcase.Name]; !ok {
+					testcaseNameToTestcase[testcase.Name] = testcase
+				}
+				continue
+			}
+			testcaseNameToTestcase[testcase.Name] = testcase
+			testcaseJSON, err := json.MarshalIndent(testcase, "", " ")
+			if err != nil {
+				log.Fatal(errors.Wrapf(err, "Couldn't marshal testsuite summary %s", testcaseJSON))
+			}
+
+			jsonFileName := fmt.Sprintf("test-%s.json", strconv.FormatInt(time.Now().UnixNano(), 10))
+			testcaseJsonFilePath := path.Join(config.ExportPath, jsonFileName)
+			if err := ioutil.WriteFile(testcaseJsonFilePath, testcaseJSON, 0644); err != nil {
+				log.Fatal(errors.Wrapf(err, "Couldn't write %s to file", testcaseJsonFilePath))
+			}
+		}
+	}
+	for _, testcase := range testcaseNameToTestcase {
+		mergedJunitXmlResult.Testcases = append(mergedJunitXmlResult.Testcases, testcase)
+	}
+	saveJunitXmlToFile(mergedJunitXmlResult)
+}
+
+func saveJunitXmlToFile(mergedJunitXmlResult *JunitXMLResult) {
+	output, err := xml.MarshalIndent(mergedJunitXmlResult, "  ", "    ")
+	if err != nil {
+		fmt.Printf("error: %v\n", err)
+	}
+	output = append([]byte(xml.Header), output...)
+
+	file, _ := os.Create(mergedJunitXmlFilePath)
+	defer file.Close()
+	if _, err = file.Write(output); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func analyzeE2eLogs(e2eLogFilePaths []string) Summary {
+	summary := Summary{TestsuiteSuccessful: false, FailedTestcases: 0, SuccessfulTestcases: 0, ExecutedTestcases: 0, TestsuiteDuration: 0, FlakedTestcases: 0, DescriptionFile: config.DescriptionFile, Flaked: false}
```

Comment:
no need to set the default values, if you initalize the struct they are already defaulted.
```suggestion
	summary := Summary{DescriptionFile: config.DescriptionFile}
```